pkg/golinters/gci.go

```diff
@@ -32,11 +33,32 @@ func NewGci() *goanalysis.Linter {
 		nil,
 	).WithContextSetter(func(lintCtx *linter.Context) {
 		localFlag := lintCtx.Settings().Gci.LocalPrefixes
+
+		if localFlag == "" && lintCtx.Settings().Gci.LocalPrefixModule {
+			var modules []string
+
+			for _, module := range lintCtx.Modules {
+				modules = append(modules, module.Path)
+			}
+
+			localFlag = strings.Join(modules, ",")
+		}
+
 		goimportsFlag := lintCtx.Settings().Goimports.LocalPrefixes
 		if localFlag == "" && goimportsFlag != "" {
 			localFlag = goimportsFlag
 		}
 
+		if localFlag == "" && lintCtx.Settings().Goimports.LocalPrefixModule {
+			var modules []string
+
+			for _, module := range lintCtx.Modules {
+				modules = append(modules, module.Path)
+			}
+
+			localFlag = strings.Join(modules, ",")
+		}
+
```

Comment:
These looks exactly the same with some duplicated code, maybe would be nice to create a function?

```go
localFlagFn := func(modules) string {
	var modules []string

	for _, module := range lintCtx.Modules {
		modules = append(modules, module.Path)
	}

	return strings.Join(modules, ",")
}
```

Then just call it twice:

```go
if localFlag == "" && lintCtx.Settings().Gci.LocalPrefixModule {
	localFlag = localFlagFn(lintCtx.Settings().Gci.LocalPrefixModule)
}

goimportsFlag := ...


if localFlag == "" && lintCtx.Settings().Goimports.LocalPrefixModule {
	localFlag = localFlagFn(lintCtx.Settings().Goimports.LocalPrefixModule)
}
```

Reply:
Absolutely!